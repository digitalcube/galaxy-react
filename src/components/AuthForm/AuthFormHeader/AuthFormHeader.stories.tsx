import React from 'react';
import { Meta, Story } from '@storybook/react';
import { AuthFormHeader, AuthFormHeaderProps } from './AuthFormHeader';
import { ImageShifterLogo } from '../../Images';

const meta: Meta = {
  title: 'Components/AuthForm/AuthFormHeader',
  component: AuthFormHeader,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<AuthFormHeaderProps> = (args) => (
  <>
    <AuthFormHeader {...args} />
  </>
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  logo: <ImageShifterLogo width="36" height="46" alt="Shifter" />,
  title: 'Log In',
  link: '/signup',
  linkText: 'sign up',
  variant: 'login',
};
