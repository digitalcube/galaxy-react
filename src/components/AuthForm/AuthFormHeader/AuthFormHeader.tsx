import React, { FC, PropsWithChildren, ReactNode } from 'react';

export type AuthFormHeaderProps = PropsWithChildren<{
  logo: ReactNode;
  title: string;
  link?: string;
  linkText?: string;
  variant: 'login' | 'signup';
}>;
export const AuthFormHeader: FC<AuthFormHeaderProps> = ({
  logo,
  title,
  link,
  linkText,
  variant,
  children,
}) => (
  <div className={`mb-4 pb-2 ${variant}-header`}>
    <div className={`mb-4 text-center ${variant}-icon`}>{logo}</div>
    <h1 className={`mb-0 font-weight-bold text-center ${variant}-title`}>
      {title}
    </h1>
    {link ? (
      <div
        className={`mt-2 font-weight-bold text-center ${variant}-switch-way`}
      >
        or <a href={link}>{linkText}</a>
      </div>
    ) : null}
    {children}
  </div>
);
