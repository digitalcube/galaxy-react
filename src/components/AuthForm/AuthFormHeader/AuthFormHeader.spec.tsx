import { render } from '@testing-library/react'
import React from 'react'
import {Default as AuthFormHeader} from './AuthFormHeader.stories'

describe('AuthFormHeader', () => {
    it('to match snapshot', () => {
        const { asFragment } = render(<AuthFormHeader {...AuthFormHeader.args as any}/>)
        expect(asFragment()).toMatchSnapshot()

    })
})