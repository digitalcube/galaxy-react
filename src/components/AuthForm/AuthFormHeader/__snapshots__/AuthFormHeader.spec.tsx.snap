// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthFormHeader to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-4 pb-2 login-header"
  >
    <div
      class="mb-4 text-center login-icon"
    >
      <img
        alt="Shifter"
        height="46"
        width="36"
      />
    </div>
    <h1
      class="mb-0 font-weight-bold text-center login-title"
    >
      Log In
    </h1>
    <div
      class="mt-2 font-weight-bold text-center login-switch-way"
    >
      or 
      <a
        href="/signup"
        to="/signup"
      >
        sign up
      </a>
    </div>
  </div>
</DocumentFragment>
`;
